docs/source/reference/yaml-spec.rst

```diff
@@ -199,6 +199,24 @@ Available fields:
       # To use a more limited but easier to manage tool:
       # https://github.com/IBM/vpc-img-inst
 
+      # Candidate resources (optional). If specified, SkyPilot will only use
+      # these candidate resources to launch the cluster. The fields specified
+      # outside of `any_of`, `ordered` will be used as the default values for
+      # all candidate resources.
+      # `any_of:` means that SkyPilot will try to find a resource that matches
+      # any of the candidate resources, i.e. the failover order will be decided
+      # by the optimizer.
+      # `ordered:` means that SkyPilot will failover through the candidate
+      # resources with the specified order.
+      # to go through 
+      any_of:
```

Comment:
should we add example like the one in failover.rst?

Reply:
Could you elaborate which example you mean?